server/src/main/java/org/elasticsearch/cluster/service/ClusterService.java

```diff
@@ -100,6 +103,16 @@ public RerouteService getRerouteService() {
         return rerouteService;
     }
 
+    public void setAllocationService(AllocationService allocationService) {
```

Comment:
I think I'd prefer to either:

1. Support injection for the reserved state plugins.
2. Add an explicit AllocationDecidersProvider object that is passed to Plugin.createComponents that can be used to access this.

Putting it here is polluting this class I think with responsibilities that does not concern it.